nupic/utils.py

```diff
@@ -76,10 +97,55 @@ def compute(slidingWindow, total, newVal, windowSize):
 
   def next(self, newValue):
     """Instance method wrapper around compute."""
-    newAverage, self.slidingWindow, self.total = self.compute(
-        self.slidingWindow, self.total, newValue, self.windowSize)
-    return newAverage
+    return self.__call__(newValue)
 
 
   def getSlidingWindow(self):
-    return self.slidingWindow
+    return list(self._history)
+
+
+#######################################
+
+class CircularBuffer():
```

Comment:
I think we cannot. Please see the comments below in description, this is a circular buffer with O(1) random access, which deque isn't. https://stackoverflow.com/questions/6427985/can-i-improve-the-speed-of-random-access-in-a-bounded-length-deque-circular-buf 

Maybe we can further optimize this CB and the MovingAverage, but it's not the goal of this PR, I just extracted the classes hidden in metrics.py